blivet/devices/storage.py

```diff
@@ -243,7 +243,7 @@ def _setTargetSize(self, newsize):
         if not isinstance(newsize, Size):
             raise ValueError("new size must of type Size")
 
-        if self.maxSize and newsize > self.maxSize:
+        if self.maxSize and newsize and newsize > self.maxSize:
```

Comment:
Anaconda then calls this:
```
    def _remove_empty_parents(self, device):
        # if this device has parents with no other children, remove them too
        for parent in device.parents:
            if parent.kids == 0 and not parent.isDisk:
                self._destroy_device(parent)
```
That looks like a quite twisted logic/thing to me, but I'm not changing this in any way.